### PR TITLE
Fix undefined dev dependency versions

### PR DESCRIPTION
**Pull request by thebriando**
# ↪️ Pull Request

At some point a bug was introduced where we are not actually resolving versions of the Parcel plugins we use.

## 💻 Examples

This is the log for one of the configs after running `yarn demo` in the `simple` example project. The versions are undefined for each devDep.

```
console: Map {
  '@parcel/transformer-react-refresh-babel' => undefined,
  '@parcel/transformer-babel' => undefined,
  '@parcel/transformer-js' => undefined,
  '@parcel/transformer-react-refresh-wrap' => undefined
}
```

## 🚨 Test instructions

Checkout the branch and add `console.log(config.devDeps)` to line 87 in `ConfigLoader.js`.
 https://github.com/thebriando/parcel/blob/10f4b4213ce0eb9234096631f64ec0a725ce73a0/packages/core/core/src/ConfigLoader.js#L87

Run `yarn parcel` on a project and ensure that there are version numbers for each devDep.

```
console: Map {
  '@parcel/transformer-react-refresh-babel' => '2.0.0-beta.1',
  '@parcel/transformer-babel' => '2.0.0-beta.1',
  '@parcel/transformer-js' => '2.0.0-beta.1',
  '@parcel/transformer-react-refresh-wrap' => '2.0.0-beta.1'
}
```

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

@padmaia @wbinnssmith

### CODE REVIEW

File: packages/core/core/src/ParcelConfig.js [hunk 1]
```diff
@@ -332,6 +422,47 @@ export default class ParcelConfig {
...
+    Array<{|
+      name: string,
+      resolved: FilePath,
+      resolveFrom: FilePath,
+      version: Semver,
+    |}>,
+  > {
+    return this.resolvePlugins(this._getResolverNodes());
+  }
+
```
Comment (padmaia) [inline packages/core/core/src/ParcelConfig.js:465]: Could you add `resolveBundler`, `resolveRuntimes`, `resolvePackager`, and `resolveOptimizers` as well? 🙏

File: packages/core/core/src/Transformation.js [hunk 2]
```diff
@@ -762,7 +762,6 @@ function getImpactfulConfigInfo(configs: ConfigMap) {
     for (let [moduleName, version] of devDeps) {
       devDepsObject[moduleName] = version;
     }
-
```
Comment (padmaia) [inline packages/core/core/src/Transformation.js:765]: (nitpick) unnecessary change

**Review by padmaia — CHANGES_REQUESTED** [review]

File: packages/core/core/src/ParcelConfig.js (hunk 1 above)
Comment (thebriando) [reply packages/core/core/src/ParcelConfig.js:537]: For these, should they also be called in `ConfigLoader.js`? Are there `actionTypes` for bundler, runtimes, packager, optimizers? 

https://github.com/parcel-bundler/parcel/blob/676499e3565d531e86178d978f89375d86fddf28/packages/core/core/src/ConfigLoader.js#L61-L75

File: packages/core/core/src/ParcelConfig.js [hunk 3]
```diff
@@ -116,23 +122,107 @@ export default class ParcelConfig {
     };
   }
 
-  loadPlugin<T>(
+  async resolvePlugin(
+    node: ParcelPluginNode,
+  ): Promise<{|resolved: FilePath, pkg?: ?PackageJSON|}> {
+    let resolved, pkg;
+    let autoinstall = this.autoinstall;
+    try {
```
[the remark this answers is not in the document]
Comment (thebriando) [reply packages/core/core/src/ParcelConfig.js:130]: Should we also store `pkg` in the cache then? Since resolve returns `pkg`
